Classes/ExportHelper.cs

```diff
@@ -1148,6 +1148,27 @@ private void ExportSegments(AppVersion ver, XElement xparent)
                         xmem.SetAttributeValue("Id", id);
                         xmem.SetAttributeValue("Address", mem.Address);
                         xmem.SetAttributeValue("Size", mem.Size);
+                        if(ver.ComObjectMemoryObject == mem){
+
+                            
+                            byte[] groupObjectTableData = CalculateGroupObjectTable(ver);
+                            byte[] segmentData = new byte[ver.ComObjectMemoryObject.Size];
+                            for(int i = 0; i < ver.ComObjectTableOffset; i++){
+                                segmentData[i] = 0;
+                            }
+                            groupObjectTableData.CopyTo(segmentData, ver.ComObjectTableOffset);
+                            if(general.Info.Mask.ManagementModel == "Bcu1"){
+                                segmentData[0] = (byte)(ver.ComObjectMemoryObject.Size - 1);
```

Comment:
For BCU1 it is very different to the other Mask Versions (BCU2, BIM112...)

The MDT_KP_BE_01_Push_Button_V15a has a mask version "MV-0701" so it is not a BCU1.
This data is calculated in function `CalculateGroupObjectTable`
I documented the content of the table here: [selfbus wiki](https://selfbus.org/wiki/software/knx-infos/55-communication-objects-management)

If the managementModel is not BCU1, the groupObject Table has its own memory and starts at offset=0.
In this case in Byte 0 there is only the amount of comObjects.
In KNX document "03_05_01 Resources v01.09.03" chapter 4.11.2.1 is described the "current size" is 1 octet (=byte).
![image](https://github.com/user-attachments/assets/3a49e668-1c08-4327-926c-cd28c82c7fb8)
So there can be 256 communication objects at maximum. But the table can be bigger in this case.

If the managementModel is BCU1 all tables are in one memory segment with an offset.
So this table has to be moved to the offset address.
In Byte 0 in absoluteSegment memory for BCU1 has to be the size of the memory section.
This is what you see in line 1161 of the code.

The code was tested with a BCU1 device and a BIM112 device. Both worked very well.

Reply:
I just want to understand it.
I dont use Absolute Memory so this is new for me.

Im reading the code so:
- You calculate the ComObjectTable for every AbsoluteMemorySegment if it is set as ComObjectSegment.
- reset all data from 0 to offset
- copy it to the segment with offset
- if it is bcu1 set length (here you dont check offset? the size belongs to the table)
- save as base64 data in segment Data

Refenecing to 4.12.2.1:
The Current Size shall indicate the number of Group Objects in the GrOT.
So "ver.ComObjectMemoryObject.Size - 1" is not right. 
More something like you did in Line 1232.

It just doesnt make sense for me^^
But if you are ok with it i will merge it

Reply:
The sequence you described is correct.

Point 4: you are right, a check if the offset + data size will fit into the array makes sense.
I will add it.
Btw: If you set the Offset of the communication table too high and calculate the memory map, it will get an error.

What you see into the chapter 4.12.2.1 is the size of the group object table. This is also for BCU1 correct and is set into line 1243: `data[0] = (byte)(numberOfComObjects); `. But in case of BCU1 this byte is moved up to the offset.

For BCU1 there is a special Byte 0 in data segment with the size of the complete memory section. 
This has nothing to do with the groupObject table.
Without this size of completet segment, the ETS will not transfer it.

So for BCU1 you have 2 sizes in the data section:
Byte 0: complete size of data segment
Byte x at offset: amount of com objects

If you have more questions or hints let me know.

Reply:
So for BCU1 the grouptable is always with a offset? (at least offset of 19)

If the complete size of the data segment is in one byte it cant be over 255. (since it would overflow in one byte)
Minus 19 bytes for bcu1 data. (Line 1161-1167) = 236 left
1 bytes for the rampointer = 235 left
3 bytes for each groupobject => 78 GroupObjects

Am i missing something?